### PR TITLE
Feat: driving stats

### PR DESCRIPTION
**Pull request by zulufoxtrot**
Driving stats, Continued work from https://github.com/Hyundai-Kia-Connect/hyundai_kia_connect_api/pull/107

It works but I'm not sure about the implementation.
I kind of hacked the current implem: used a dataclass to store a day, then put that object into the dict returned by _get_driving_info. Not very clean but I don't see how to do it better without refactoring the whole thing.

I also don't know if/how it works with other regions. Tested on Kia EU only.

Feel free to suggest changes.

Preview of the data on Grafana:

image

### CODE REVIEW

**Comment by cdnninja** [comment]
This is awesome work!  Really appreciate it.  I added a few comments.

**Comment by zulufoxtrot** [comment]
Thank you. I can't see your comments, maybe you need to submit your review?

**Review by cdnninja — CHANGES_REQUESTED** [review]

File: hyundai_kia_connect_api/KiaUvoApiEU.py [hunk 1]
```diff
@@ -180,6 +180,8 @@ def get_last_updated_at(self, value) -> dt.datetime:
     def update_vehicle_with_cached_state(self, token: Token, vehicle: Vehicle) -> None:
         state = self._get_cached_vehicle_state(token, vehicle)
         self._update_vehicle_properties(vehicle, state)
+        state = self._get_driving_info(token, vehicle)
```
[the remark this answers is not in the document]
Comment (zulufoxtrot) [reply hyundai_kia_connect_api/KiaUvoApiEU.py]: Got it.
- added a try except block
- removed the try except block from _get_driving_info, it's better to have it IMO.

File: hyundai_kia_connect_api/Vehicle.py [hunk 2]
```diff
@@ -15,6 +15,20 @@ class EvChargeLimits:
...
+class DailyDrivingStats:
+    # energy stats are expressed in watthours (Wh)
+    date: datetime.datetime = None
+    total_consumed: int = None
+    engine_consumption: int = None
+    climate_consumption: int = None
+    onboard_electronics_consumption: int = None
+    battery_care_consumption: int = None
+    regenerated_energy: int = None
+    # distance is expressed in (I assume) whatever unit the vehicle is configured in. KMs (rounded) in my case
```
[the remark this answers is not in the document]
Comment (cdnninja) [reply hyundai_kia_connect_api/Vehicle.py:28]: I am thinking we use a default for now.   Also happy to learn from you on best practices.  This is just a hobby for me you seem to have more best practice experience.

[the remark this answers is not in the document]
Comment (zulufoxtrot) [reply hyundai_kia_connect_api/Vehicle.py:28]: OK! I'll try to update this PR asap.

File: hyundai_kia_connect_api/KiaUvoApiEU.py [hunk 3]
```diff
@@ -354,6 +356,11 @@ def _update_vehicle_properties(self, vehicle: Vehicle, state: dict) -> None:
             )
         vehicle.data = state
 
+    def _update_vehicle_drive_info(self, vehicle: Vehicle, state: dict) -> None:
+        vehicle.total_power_consumed = get_child_value(state, "totalPwrCsp")
```
[the remark this answers is not in the document]
Comment (zulufoxtrot) [reply hyundai_kia_connect_api/KiaUvoApiEU.py:412]: It's watt-hours (Wh). The units are not given explicitly by the API on this endpoint.
Contrarily to distances/temperatures, which can depend on the region, I think energy & power values will always be expressed in Wh or kWh.

Based on this assumption, I suggest to:
- not define a unit, and remove the following attributes: `_total_power_consumed_unit`, `_power_consumption_30d_unit`, `_total_power_consumed_value`, `_power_consumption_30d_value`
- document the remaining attributes: `_total_power_consumed`, `_power_consumption_30d`. Add a comment speciflying the unit
- make these 2 attributes public (remove underscore prefix) and remove their getters and setters, for simplification

What do you think?

[the remark this answers is not in the document]
Comment (cdnninja) [reply hyundai_kia_connect_api/KiaUvoApiEU.py:412]: I'm good with this! If we find another region uses a different unit we can fix at that time.

[the remark this answers is not in the document]
Comment (cdnninja) [reply hyundai_kia_connect_api/KiaUvoApiEU.py:412]: Let me know when you are ready for me to merge.

File: hyundai_kia_connect_api/Vehicle.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (zulufoxtrot) [reply hyundai_kia_connect_api/Vehicle.py:28]: distance unit attribute added, set to KM by default

File: hyundai_kia_connect_api/KiaUvoApiEU.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (zulufoxtrot) [reply hyundai_kia_connect_api/KiaUvoApiEU.py:412]: changes implemented as discussed.

[the remark this answers is not in the document]
Comment (zulufoxtrot) [reply hyundai_kia_connect_api/KiaUvoApiEU.py:412]: If you're OK with the result, you can merge.

**Pull request merged**